src/guiguts/tools/pptxt.py

```diff
@@ -0,0 +1,6738 @@
+                if heading in specials_report:
+                    specials_report[heading].append(
+                        make_a_report_record(line_index + 1, line)
+                    )
+                else:
+                    specials_report[heading] = [
+                        make_a_report_record(line_index + 1, line)
+                    ]
+
+            if "&" in line and "&#" not in line:
+                heading = "\nampersand character in line (excluding unicode numeric character references)"
+                if heading in specials_report:
+                    specials_report[heading].append(
+                        make_a_report_record(line_index + 1, line)
+                    )
+                else:
+                    specials_report[heading] = [
+                        make_a_report_record(line_index + 1, line)
+                    ]
+
+            lc_line = line.lower()
+            if re_comma.search(lc_line):
+                heading = "\nunexpected comma after certain words"
+                if heading in specials_report:
+                    specials_report[heading].append(
+                        make_a_report_record(line_index + 1, line)
+                    )
+                else:
+                    specials_report[heading] = [
+                        make_a_report_record(line_index + 1, line)
+                    ]
+
+            if re_period.search(lc_line):
+                heading = "\nunexpected period after certain words"
+                if heading in specials_report:
+                    specials_report[heading].append(
+                        make_a_report_record(line_index + 1, line)
+                    )
+                else:
+                    specials_report[heading] = [
+                        make_a_report_record(line_index + 1, line)
+                    ]
+
+            line_index += 1
+
+        # We have at this point a dictionary of headings and accompanying line
+        # text, etc., that are formatted as complete records ready to be written
+        # to the output file. So just output the whole dictionary.
+
+        for heading in sorted(specials_report):
+            print(heading, file=string_out)
+            for record in specials_report[heading]:
+                print(record, file=string_out)
+
+        # We may get here without finding any 'specials' entries.
+        if len(specials_report) == 0:
+            print("\n    no special situations reports.", file=string_out)
+
+        # All done! Write the rule that terminates this section of the report.
+        template = "\n{}"
+        print(template.format("-" * 80), file=string_out)
+
+    def build_scanno_dictionary() -> Dict[str, int]:
+        """Builds a dictionary from a list of common misspelled words."""
+        # List of common scannos. Add additions to end of the list.
+        # Duplicates will be dealt with when building the dictionary.
+
+        scannos_misspelled = [
```

Comment:
Although it is a bit unwieldy, I wonder if that's a relatively small price to pay for the developers, compared to the potential pain for users of a sidecar file. That's only because we want this to work as a standalone script as well as a module imported into GG, or a separate scannos file would be a no-brainer. 
The "bin" sidecar files associated with text files that GG uses, are a nuisance that do cause issues when users copy/move files, and I would prefer to avoid similar issues with this.
I suggest we keep the list of scannos in the code, at least for now. It can be revisited later, especially if we find that other tools would benefit from sharing the scannos list. Would you be ok with that @cpeel ?
I think @G4OEU has already put the list as far down the source file as possible - I presume there's no sensible way of putting it right at the bottom (out of sight 😉)

Reply:
That makes sense to me. If we wanted to increase the density of this and still make black happy we can wrap it in `# fmt: off` / `# fmt:on` tags and shave ~3k lines off the file, eg: https://github.com/cpeel/Guiguts-2.0-tkinter/commit/6472608c44ff3823ca59dece59db0f95443e1314

Reply:
😳 Using `fmt: off` and multiple words per line is how @G4OEU originally had it, and I said "I'd prefer not to use directives  that alter black's behavior if not really needed", so he reverted to the 3000 extra lines version. Since both of you have had thoughts along those lines, however, I'm happy for that to be done, if Quentin is happy to re-do it either as he had it originally (more aligned) or with Casey's version (more dense).